lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosResult.cxx

```diff
@@ -307,6 +313,110 @@ void FunctionalChaosResult::setInvolvesModelSelection(const Bool involvesModelSe
   involvesModelSelection_ = involvesModelSelection;
 }
 
+/* Conditional expectation accessor */
+FunctionalChaosResult FunctionalChaosResult::getConditionalExpectation(const Indices & conditioningIndices) const
+{
+  // Get marginal input sample and distribution
+  const UnsignedInteger inputDimension = inputSample_.getDimension();
+  conditioningIndices.check(inputDimension);
+  const Sample inputSampleMarginal(inputSample_.getMarginal(conditioningIndices));
+  const Distribution inputDistributionMarginal(distribution_.getMarginal(conditioningIndices));
+  
+  // Check independence
+  if (!distribution_.hasIndependentCopula())
+    throw InvalidArgumentException(HERE) << "This class can only manage an independent copula.";
```

Comment:
I propose a more detailed error message:
FunctionalChaosResult can only compute the conditional expectation for independent copulas

Reply:
Ok. I updated the message in 271a037da58754890130951f83d7f7b33635a80c